TwitchPlaysAssembly/Src/ComponentSolvers/Modded/Shims/VexillologyShim.cs

```diff
@@ -0,0 +1,120 @@
+		int FlagColour1 = _component.GetValue<int>("FlagColour1");
+		int FlagColour2 = _component.GetValue<int>("FlagColour2");
+		int FlagColour3 = _component.GetValue<int>("FlagColour3");
+		int ActiveColour = _component.GetValue<int>("ActiveColour");
+		int SubmitTime = _component.GetValue<int>("SubmitTime");
+		if (AnswerColour1 != FlagColour1)
+		{
+			if (ActiveColour != AnswerColour1)
+			{
+				ActiveColour = AnswerColour1;
+				yield return DoInteractionClick(_cButtons[ActiveColour]);
+			}
+			switch (ActiveFlag)
+			{
+				case 0:
+					yield return DoInteractionClick(_horizFlag[0]);
+					break;
+				case 1:
+					yield return DoInteractionClick(_vertFlag[0]);
+					break;
+				case 2:
+					yield return DoInteractionClick(_sweFlag[UnityEngine.Random.Range(0, 4)]);
+					break;
+				default:
+					yield return DoInteractionClick(_norFlag[UnityEngine.Random.Range(0, 4)]);
+					break;
+			}
+		}
+		if (AnswerColour2 != FlagColour2)
+		{
+			if (ActiveColour != AnswerColour2)
+			{
+				ActiveColour = AnswerColour2;
+				yield return DoInteractionClick(_cButtons[ActiveColour]);
+			}
+			switch (ActiveFlag)
+			{
+				case 0:
+					yield return DoInteractionClick(_horizFlag[1]);
+					break;
+				case 1:
+					yield return DoInteractionClick(_vertFlag[1]);
+					break;
+				case 2:
+					yield return DoInteractionClick(_sweFlag[UnityEngine.Random.Range(4, 6)]);
+					break;
+				default:
+					yield return DoInteractionClick(_norFlag[UnityEngine.Random.Range(4, 6)]);
+					break;
+			}
+		}
+		if (AnswerColour3 != FlagColour3 && ActiveFlag != 2)
+		{
+			if (ActiveColour != AnswerColour3)
+			{
+				ActiveColour = AnswerColour3;
+				yield return DoInteractionClick(_cButtons[ActiveColour]);
+			}
+			switch (ActiveFlag)
+			{
+				case 0:
+					yield return DoInteractionClick(_horizFlag[2]);
+					break;
+				case 1:
+					yield return DoInteractionClick(_vertFlag[2]);
+					break;
+				default:
+					yield return DoInteractionClick(_norFlag[UnityEngine.Random.Range(6, 14)]);
+					break;
+			}
+		}
+		if (_component.GetValue<bool>("_ChadRomania"))
+			while ((edgework.GetTime() >= 60 && !edgework.GetFormattedTime().Contains('0') && !edgework.GetFormattedTime().Contains('5')) || (edgework.GetTime() < 60 && !edgework.GetFormattedTime().Remove(2).Contains('0') && !edgework.GetFormattedTime().Remove(2).Contains('5'))) yield return true;
+		else if (SubmitTime != 10)
+			while ((edgework.GetTime() >= 60 && !edgework.GetFormattedTime().Contains(SubmitTime.ToString()[0])) || (edgework.GetTime() < 60 && !edgework.GetFormattedTime().Remove(2).Contains(SubmitTime.ToString()[0]))) yield return true;
+		else
+			while ((edgework.GetTime() >= 60 && !edgework.GetFormattedTime().Contains(edgework.GetSerialNumberNumbers().Last().ToString()[0])) || (edgework.GetTime() < 60 && !edgework.GetFormattedTime().Remove(2).Contains(edgework.GetSerialNumberNumbers().Last().ToString()[0]))) yield return true;
```

Comment:
This felt too messy, so I tried my best to simplify it. Let me know if I made any mistakes.
```suggestion

		char[] digits = null;
		if (_component.GetValue<bool>("_ChadRomania"))
			digits = new[] { '0', '5' };
		else if (SubmitTime != 10)
			digits = new[] { SubmitTime.ToString()[0] };
		else
			digits = new[] { edgework.GetSerialNumberNumbers().Last().ToString()[0] };

		while (true)
		{
			var time = edgework.GetTime() >= 60 ? edgework.GetFormattedTime() : edgework.GetFormattedTime().Remove(2);
			if (digits.Any(time.Contains))
				break;

			yield return true;
		}

```